app/types/value_object_type.rb

```diff
@@ -7,12 +7,20 @@ def cast(value)
       value.map do |data|
         next data unless data.is_a?(Hash)
 
-        klass = data['class'].constantize
-        klass.find_by_key(data['key'])
+        cast_to_value_object(data)
       end
+    when Hash
+      cast_to_value_object(value)
+    when ValueObject
+      value
     end
   end
 
+  def cast_to_value_object(data)
+    klass = data['class'].constantize
```

Comment:
Due to the suggested changes we will receive both `String` and `Symbol` keys, and need to handle this here. Otherwise: Very nice improvement 🎉

Reply:
Cool cool cool. See 685e7a6